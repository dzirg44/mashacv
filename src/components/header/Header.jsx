import React from 'react';
import logo from './image.jpg';

const Header = () => (
  <header>
    <img src={logo} alt="logo" width="200px" />
    <h2>Mariia Neymorovets</h2>
    <h3>PHP Back-end developer</h3>
  </header>
);

export { Header as default };
