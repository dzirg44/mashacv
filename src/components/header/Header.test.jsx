import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('<Header />', () => {
  it('should be defined', () => {
    expect(Header).toBeDefined();
  });

  it('renders correctly', () => {
    const header = shallow(<Header />);

    expect(header).toMatchSnapshot();
  });

  it('renders img', () => {
    const header = shallow(<Header />);

    expect(header.find('img').length).toBeGreaterThan(0);
  });

  it('renders h2', () => {
    const header = shallow(<Header />);
    const text = <h2>Mariia Neymorovets</h2>;

    expect(header.contains(text)).toEqual(true);
  });

  it('renders h3', () => {
    const header = shallow(<Header />);
    const text = <h3>PHP Back-end developer</h3>;

    expect(header.contains(text)).toEqual(true);
  });
});
