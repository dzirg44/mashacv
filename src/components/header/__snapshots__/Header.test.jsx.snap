// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="logo"
          src="image.png"
          width="200px"
/>,
        <h2>
          Masha
</h2>,
        <h3>
          PHP Developer
</h3>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "logo",
          "src": "image.png",
          "width": "200px",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Masha",
        },
        "ref": null,
        "rendered": "Masha",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "PHP Developer",
        },
        "ref": null,
        "rendered": "PHP Developer",
        "type": "h3",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            src="image.png"
            width="200px"
/>,
          <h2>
            Masha
</h2>,
          <h3>
            PHP Developer
</h3>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "src": "image.png",
            "width": "200px",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Masha",
          },
          "ref": null,
          "rendered": "Masha",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "PHP Developer",
          },
          "ref": null,
          "rendered": "PHP Developer",
          "type": "h3",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
